packages/core/pluggableElementTypes/models/BaseDisplayModel.tsx

```diff
@@ -36,7 +36,10 @@ export const BaseDisplay = types
     },
 
     get adapterConfig() {
-      return getConf(this.parentTrack, 'adapter')
+      const conf = getConf(this.parentTrack, 'adapter')
+      return conf.type
```

Comment:
this one is a little weird. does this maybe mean there are places that should be using getConf(obj,'type') but instead are using obj.type?

Reply:
This is to handle a corner case where the adapter config matches the default config, and so the adapter snapshot just ends up as an empty object. The only time I've seen this happen so far is a fromConfigAdapter with no features.

Reply:
this is an interesting one. one possible alternative here: if we are really returning a config object, maybe the code should actually be this?

```
diff --git a/packages/core/pluggableElementTypes/models/BaseDisplayModel.tsx b/packages/core/pluggableElementTypes/models/B>
index ccb90a996..cd8a42bbe 100644
--- a/packages/core/pluggableElementTypes/models/BaseDisplayModel.tsx
+++ b/packages/core/pluggableElementTypes/models/BaseDisplayModel.tsx
@@ -36,10 +36,7 @@ export const BaseDisplay = types
     },

     get adapterConfig() {
-      const conf = getConf(this.parentTrack, 'adapter')
-      return conf.type
-        ? conf
-        : { type: getConf(this.parentTrack, ['adapter', 'type']) }
+      return this.parentTrack.configuration.adapter
     },


```

I think getConf snapshots the config, so by doing this instead, it keeps downstream code able to use getConf appropriately

Reply:
there are some weird bugs that come up if we use my suggested diff...see https://github.com/GMOD/jbrowse-components/runs/2959561046#step:5:3444

we could potentially fix those or accept this PR as is. I could go either way :)

Reply:
I ended up reverting this to the way it was before this PR, since I was able to handle this corner case in a better way (see below), so there's no longer a change in this file. We could see about avoiding the getConf sometime, but that would be a separate change.